Controllers/ShipmentController.cs

```diff
@@ -17,16 +19,21 @@ public class ShipmentController : ControllerBase
         private readonly IShipmentType _shipmentType;
         private readonly IShipment _shipment;
         private readonly IMapper _mapper;
+        private readonly IShipmentInternalDataClient _http;
+        private readonly ITransactionInternal _transactionInternal;
 
-        public ShipmentController(IShipmentType shipmentType, 
-            IShipment shipment, IMapper mapper)
+        public ShipmentController(IShipmentType shipmentType,
+            IShipment shipment, IMapper mapper, IShipmentInternalDataClient http,
+            ITransactionInternal transactionInternal)
         {
             _shipmentType = shipmentType;
             _shipment = shipment;
             _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _http = http; //buat call method di httprequest (?)
+            _transactionInternal = transactionInternal;
         }
 
-        [HttpPost]
+        [HttpPost("GetShipmentFee")]
```

Comment:
new route

Reply:
```suggestion
        [HttpPost("fee")]
```

mending gini aja, nanti bentuk url nya jadi `api/v1/shipment/fee`